realm/rfc/0001-backlink-query-syntax.md

```diff
@@ -0,0 +1,232 @@
+- Feature Name: Inverse relationship syntax
+- Start Date: 2016-05-30
+- RFC PR: 
+- Version: 3
+
+# Summary
+
+It should be possible to define and query backlinks in both `Realm` and 
+`DynamicRealm`. As a `DynamicRealm` can work without _any_ Realm model 
+classes, it should be possible to query backlinks without them being defined there.
+
+# Motivation
+
+Backlinks are special in Realm. They are not defined explicitly in the schema,
+but are automatically created when defining Links and LinkLists (forward links).
+
+This creates some conflicts with the current query system as it rely on all 
+fields being defined by the schema. This is called *named backlinks*.
+
+This proposal introduces a new syntax for queries so it is possible to query
+across a backlink even though it hasn't been defined by a model class. This is
+called *unnamed backlinks*.
```

Comment:
To me those seem like cases where it would be _nice_ to support querying over unnamed backlinks, but they're much less important than supporting named backlinks. I'm curious why you feel they're important enough to warrant tackling in your initial implementation? On the Realm Cocoa side we concluded that they weren't necessary for our initial implementation, and we wanted to avoid inventing special query syntax to accommodate them at this point.